### ISSUE TITLE
Explicit Newlines with \ at the end of the line

### ISSUE THREAD

**Issue by Treeniks** [retypeapp/retype#167]
Currently, the only explicit way to get retype to add a newline, but not a new paragraph, is by having two spaces at the end of the line. This is expected and correct markdown behaviour, however another way to add explicit newlines in markdown is usually by adding a `\` at the end of the line. Currently, retype ignores this.

This would not add any functionality, nor is a trailing `\` even part of most markdown standards (at least not to my knowledge), however there is two reasons to still add this feature:
- a `\` at the end of the line is much more explicit than two spaces, because it's not invisible
- some people use "remove trailing whitespace" features with their code editors, thus adding two trailing spaces becomes a huge hassle

I would really like to see this added, because this is a common occurance for me and I do happen to have "remove trailing whitespace" enabled in basically all my editors.

**Comment by geoffreymcgill** [comment]
To the best of my knowledge, the trailing `\` backslash is not part of the CommonMark spec, but it is supported by Retype.

The following sample works for me:

```md
# Line endings

Paragraph one\
Paragraph two\
Paragraph three
```

<img width="297" alt="Screen Shot 2021-10-05 at 8 43 49 AM" src="https://user-images.githubusercontent.com/62210/136026334-ae64b9f4-387c-4f0e-b381-19de1fab00a5.png">

Without the trailing `\`, no line breaks are added:

```md
# Line endings

Paragraph one
Paragraph two
Paragraph three
```

<img width="427" alt="Screen Shot 2021-10-05 at 8 53 35 AM" src="https://user-images.githubusercontent.com/62210/136026608-43a40b93-1a20-4847-88e7-0f155b86f12a.png">

@Treeniks Can you test the sample above and confirm if it is working (or not) as per your requirements? Please ensure you are using the latest Retype release (`v1.10.0`).

Related to this topic, if you would like to switch between the CommonMark soft and hard line breaks, Retype has the [`markdown.lineBreaks`](https://retype.com/configuration/project/#linebreaks) project config. Default is `soft`.

Switching to `hard` line breaks would require adding the following config to your projects `retype.yml`:

```md
markdown:
  lineBreaks: hard
```

Hope this helps.

**Comment by Treeniks** [comment]
I tested the sample and it does not work for me, it looks like this instead:
![Screenshot 2021-10-05 161743](https://user-images.githubusercontent.com/56131826/136041301-cba02b60-328f-4745-b4c0-6a52f645995a.png)
but I figured out that it seems to be a problem with Unix vs Windows line endings. When I switch my source Markdown file to Unix line endings, the sample does work as expected, but with Windows line endings it does not.

**Comment by fabriciomurta** [comment]
Looks like we could fix the issue just by upgrading our under-the-hood markdown processor. That's going to be in the next release!

**Comment by geoffreymcgill** [comment]
Hi @Treeniks. As mentioned above, we were able to isolate the issue and found an easy fix. The fix will be included in the next release (`v1.11`). At the moment, there is not a fixed release date for `v1.11`, but we do our best to get it released as soon as possible, hopefully inside October.

If this issue is critical to your project, send me an email (geoff@object.net) and we should be able to send you a `-preview` build with the fix.

**Comment by geoffreymcgill** [comment]
@Treeniks Retype [v1.11.0](https://github.com/retypeapp/retype/releases/tag/v1.11.0) is now available.
